src/fetch_helpers.jl

```diff
@@ -14,18 +14,50 @@ function progress_update_period()
             # Running in a script, probably want minimal updates
             "Inf"
         else # default
-            if haskey(ENV, "DATADEP_PROGRESS_UPDATE_PERIOD")
-                @warn "\"DATADEP_PROGRESS_UPDATE_PERIOD\" is deprecated, use \"DATADEPS_PROGRESS_UPDATE_PERIOD\" instead."
-                get(ENV, "DATADEP_PROGRESS_UPDATE_PERIOD", "5")
-            else
-                "5" # seconds
-            end
+            "5" # seconds
         end
     end
-    parse(Float32, envvar) 
+    parse(Float32, envvar)
 end
 
+"""
+    fetch_default(remote_path, local_path)
+
+The default fetch method.
+It tries to be a little bit smart to work with things other than just HTTP.
+See also [`fetch_base`](@ref) and [`fetch_http`](@ref).
+"""
+function fetch_default(remotepath, localdir)
+    if remotepath isa AbstractString && occursin(r"^https?://", remotepath)
```

Comment:
Maybe include a link to the discussion on base as to why these are different?

https://github.com/JuliaLang/julia/pull/33088

Reply:
Not really relevent I think.
Real difference is wither we need to use `Base.basename` or that we have a download that can get it right on its own.
Not so much about joinpath stuff